src/triedb.rs

```diff
@@ -147,6 +149,56 @@ impl<'db> Trie for TrieDB<'db> {
     }
 }
 
+impl<'db> CryptoStructure for TrieDB<'db> {
+    fn make_proof(&self, key: &H256) -> crate::Result<(CryptoProofUnit, CryptoProof)> {
+        type Unit = CryptoProofUnit;
+
+        fn make_proof_upto<'k>(
+            db: &'k dyn HashDB,
+            path: &'k NibbleSlice<'_>,
+            hash: &'k H256,
+        ) -> crate::Result<(Option<Bytes>, Vec<Bytes>)> {
+            let node_rlp = db.get(&hash).ok_or_else(|| TrieError::IncompleteDatabase(*hash))?;
+
+            match Node::decoded(&node_rlp) {
+                Some(Node::Leaf(partial, value)) => {
+                    if &partial == path {
+                        Ok((Some(value.to_vec()), vec![node_rlp]))
+                    } else {
+                        Ok((None, vec![node_rlp]))
+                    }
+                }
+                Some(Node::Branch(partial, children)) => {
+                    if path.starts_with(&partial) {
+                        let left_path = path.mid(partial.len() + 1);
+                        match children[path.mid(partial.len()).at(0) as usize] {
+                            Some(x) => {
+                                let r = make_proof_upto(db, &left_path, &x)?;
+                                Ok((r.0, [vec![node_rlp], r.1].concat()))
+                            }
+                            None => Ok((None, vec![node_rlp])),
+                        }
+                    } else {
+                        Ok((None, Vec::new()))
+                    }
+                }
+                None => Ok((None, Vec::new())), // empty trie
+            }
+        }
+        let x = make_proof_upto(self.db, &NibbleSlice::new(&key), self.root())?;
```

Comment:
```rust
        let (value, proof) = make_proof_upto(self.db, &NibbleSlice::new(&key), self.root())?;
        let unit = Unit {
            hash: *self.root(),
            key: *key,
            value,
        };
        Ok((unit, CryptoProof(proof)))
```
x.0 is already an `Option<Bytes>`.
and you don't need `0: ...` to create a new type.